### ISSUE TITLE
[FEATURE] embedding layer

### ISSUE THREAD

**Issue by ohbigkite** [golansformer/golansformer#11]
## Feature Description

해당 Issue는 transformer의 embedding layer를 구현합니다.

## Motivation and Context

embedding layer는 transformer 구조만이 아니라 다양한 nlp 모델에 입력으로 사용되는 기초적인 아키텍쳐입니다.
embedding layer는 단어를 밀집 표현으로 변환해 벡터로 표현하며, 초기화된 임베딩은 학습 과정을 통해 조정되어 단어들 사이의 의미론적 유사성을 반영하게 됩니다. transformer에서는 embedding layer 이후 positional encoding layer와 더해져서 encoder의 입력으로 사용됩니다. 
pytorch에서는 nn.Embedding()으로 간단하게 구현되는 부분을 직접 구현했습니다. 

## Proposed Solution

임베딩 층의 입력으로 사용하기 위해서 입력 시퀀스의 각 단어들은 모두 정수 인코딩이 되어있다고 가정했습니다.

- [x] Embedding 구조체 정의 : 임베딩 정보를 저장하기 위한 기본 구조체를 정의, 2차원 float32 slice 
- [x] NewEmbedding 함수 정의 : 임베딩 구조체를 생성하고 초기화하는 생성자(constructor) 역할을 하는 함수, 임베딩할 단어의 개수와 임베딩 벡터의 차원을 입력으로 받습니다.
- [x] 임의의 값으로 벡터 초기화 : rand.Float32()함수를 사용해 0과 1사이 임의의 값으로 초기화 됨.

## Alternatives Considered

nn.Embedding() 함수에는 padding 또한 고려하는 부분이 있으나 해당 부분은 optional한 부분이기 때문에 추후 개선하도록 하겠습니다.

## Additional Information

추후 develop 할 부분
- [ ] positional encoding part와 연결되어 encoding layer로 들어가는 코드 작성하기
